autoload/neomake/makers/ft/rust.vim

```diff
@@ -22,3 +22,70 @@ function! neomake#makers#ft#rust#rustc() abort
             \ '%-G%s',
         \ }
 endfunction
+
+function! neomake#makers#ft#rust#cargo() abort
+    return {
+        \ 'args': ['test', '--no-run', '--message-format=json', '--quiet'],
+        \ 'append_file': 0,
+        \ 'errorformat':
+            \ '[%t%n] "%f" %l:%v %m,'.
+            \ '[%t] "%f" %l:%v %m',
+        \ 'process_output': function('neomake#makers#ft#rust#CargoProcessOutput'),
+        \ }
+endfunction
+
+function! neomake#makers#ft#rust#CargoProcessOutput(context) abort
+    let l:errors = []
+    for l:line in a:context['output']
+        if l:line[0] !=# '{'
+            continue
+        endif
```

Comment:
This does not work in case cargo would not provide a json dump per line, right?
Shouldn't you join all lines/output, and then decode those?